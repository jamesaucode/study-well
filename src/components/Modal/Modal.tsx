import React, { useEffect, EffectCallback } from "react";
import ReactDOM from "react-dom";
import styled from "styled-components";
import SVG from "react-inlinesvg";
import { fadeIn } from "../../../utils/style";
import { NextFC } from "next";

const ModalWrapper = styled.div`
  align-items: center;
  animation: ${fadeIn} 0.5s ease-out 1;
  background: #fff;
  border-radius: 3px;
  bottom: 0;
  box-shadow: 4px 4px 8px rgba(0, 0, 0, 0.25);
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  height: fit-content;
  justify-content: center;
  left: 0;
  margin: auto;
  max-height: 600px;
  min-height: 300px;
  padding: 2rem;
  position: absolute;
  right: 0;
  top: 0;
  width: fit-content;
  z-index: 100;
`;
const DivOverlay = styled.div`
  background: rgba(0, 0, 0, 0.5);
  height: 100vh;
  position: fixed;
  top: 0;
  width: 100vw;
  z-index: 10;
`;
const Wrapper = styled.div`
  align-items: "center";
  display: "flex";
  justify-content: "center";
`;
const CrossWrapper = styled.div`
  & > span > svg {
    padding: 1rem;
    &:hover {
      cursor: pointer;
    }
  }
`;
interface ModalProps {
  closeModal: (value: void) => void;
  parentProps?: Record<string, any>;
}
export const Modal: NextFC<ModalProps> = ({
  closeModal,
  children,
}): JSX.Element => {
  // For testing
  const handleKeyDown = (event: KeyboardEvent): void => {
    if (event.keyCode === 27) {
      closeModal();
    }
  };
  useEffect((): EffectCallback => {
    console.log("Modal Mounted");
    window.addEventListener("keydown", handleKeyDown);
    // Cleanup function
    return (): void => {
      window.removeEventListener("keydown", handleKeyDown);
      console.log("Unmounting");
    };
  }, [handleKeyDown]);
  const handleClick = (): void => {
    closeModal();
  };
  return ReactDOM.createPortal(
    <>
      <DivOverlay onClick={handleClick} />
      <Wrapper>
        <ModalWrapper>
          <CrossWrapper onClick={handleClick}>
            <SVG
              className="small-icon top-right"
              src="/static/images/close_black.svg"
            />
          </CrossWrapper>
          {children}
        </ModalWrapper>
      </Wrapper>
    </>,
    document.body,
  );
};

Modal.defaultProps = {
  closeModal: (): void => {
    console.log("No function was passed in..");
  },
};
