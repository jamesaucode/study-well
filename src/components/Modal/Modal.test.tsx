/* eslint-disable */
import React from "react";
import { mount } from "enzyme";
import Modal from "./index";

describe("Modal Tests", () => {
  it("Component Renders", () => {
    const wrapper = mount(<Modal closeModal={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
