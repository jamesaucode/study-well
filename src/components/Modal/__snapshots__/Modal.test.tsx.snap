// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Tests Component Renders 1`] = `
<Modal
  closeModal={[Function]}
>
  <Portal
    containerInfo={
      <body>
        <div
          class="Modal__DivOverlay-pdadxn-1 iSuvMu"
        />
        <div
          class="Modal__Wrapper-pdadxn-2 imopbv"
        >
          <div
            class="Modal__ModalWrapper-pdadxn-0 lckyPw"
          >
            <div
              class="Modal__CrossWrapper-pdadxn-3 dIKJxW"
            >
              <span
                class="isvg loading small-icon top-right"
              />
            </div>
          </div>
        </div>
      </body>
    }
  >
    <Modal__DivOverlay
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "Modal__DivOverlay-pdadxn-1",
              "isStatic": true,
              "lastClassName": "iSuvMu",
              "rules": Array [
                "background:rgba(0,0,0,0.5);height:100vh;position:fixed;top:0;width:100vw;z-index:10;",
              ],
            },
            "displayName": "Modal__DivOverlay",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "Modal__DivOverlay-pdadxn-1",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[Function]}
      >
        <div
          className="Modal__DivOverlay-pdadxn-1 iSuvMu"
          onClick={[Function]}
        />
      </StyledComponent>
    </Modal__DivOverlay>
    <Modal__Wrapper>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "Modal__Wrapper-pdadxn-2",
              "isStatic": true,
              "lastClassName": "imopbv",
              "rules": Array [
                "align-items:\\"center\\";display:\\"flex\\";justify-content:\\"center\\";",
              ],
            },
            "displayName": "Modal__Wrapper",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "Modal__Wrapper-pdadxn-2",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="Modal__Wrapper-pdadxn-2 imopbv"
        >
          <Modal__ModalWrapper>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "Modal__ModalWrapper-pdadxn-0",
                    "isStatic": true,
                    "lastClassName": "lckyPw",
                    "rules": Array [
                      "align-items:center;animation:",
                      Keyframes {
                        "id": "sc-keyframes-eMLfYp",
                        "inject": [Function],
                        "name": "eMLfYp",
                        "rules": Array [
                          "@-webkit-keyframes eMLfYp{from{opacity:0;}to{opacity:1;}}",
                          "@keyframes eMLfYp{from{opacity:0;}to{opacity:1;}}",
                        ],
                        "toString": [Function],
                      },
                      " 0.5s ease-out 1;background:#fff;border-radius:3px;bottom:0;box-shadow:4px 4px 8px rgba(0,0,0,0.25);box-sizing:border-box;display:flex;flex-direction:column;height:fit-content;justify-content:center;left:0;margin:auto;max-height:600px;min-height:300px;padding:2rem;position:absolute;right:0;top:0;width:fit-content;z-index:100;",
                    ],
                  },
                  "displayName": "Modal__ModalWrapper",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "Modal__ModalWrapper-pdadxn-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="Modal__ModalWrapper-pdadxn-0 lckyPw"
              >
                <Modal__CrossWrapper
                  onClick={[Function]}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Modal__CrossWrapper-pdadxn-3",
                          "isStatic": true,
                          "lastClassName": "dIKJxW",
                          "rules": Array [
                            "& > span > svg{padding:1rem;&:hover{cursor:pointer;}}",
                          ],
                        },
                        "displayName": "Modal__CrossWrapper",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Modal__CrossWrapper-pdadxn-3",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                  >
                    <div
                      className="Modal__CrossWrapper-pdadxn-3 dIKJxW"
                      onClick={[Function]}
                    >
                      <InlineSVG
                        baseURL=""
                        cacheGetRequests={false}
                        className="small-icon top-right"
                        onLoad={[Function]}
                        src="/static/images/close_black.svg"
                        supportTest={[Function]}
                        uniquifyIDs={true}
                        wrapper={[Function]}
                      >
                        <span
                          className="isvg loading small-icon top-right"
                        />
                      </InlineSVG>
                    </div>
                  </StyledComponent>
                </Modal__CrossWrapper>
              </div>
            </StyledComponent>
          </Modal__ModalWrapper>
        </div>
      </StyledComponent>
    </Modal__Wrapper>
  </Portal>
</Modal>
`;
